File: .tmp_py4013/batch_04_sources/2497_maximum_star_sum_of_a_graph.js
// LeetCode 2497 - Maximum Star Sum of a Graph
// https://leetcode.com/problems/maximum-star-sum-of-a-graph/

/**
 * @param {number[]} vals
 * @param {number[][]} edges
 * @param {number} k
 * @return {number}
 */
var maxStarSum = function(vals, edges, k) {
    const n = vals.length;
    const g = Array.from({ length: n }, () => []);
    for (const [a, b] of edges) {
        g[a].push(b);
        g[b].push(a);
    }
    let ans = vals[0];
    for (let i = 0; i < n; i++) {
        const neigh = [];
        for (const v of g[i]) {
            if (vals[v] > 0) neigh.push(vals[v]);
        }
        neigh.sort((a, b) => b - a);
        let sum = vals[i];
        for (let j = 0; j < neigh.length && j < k; j++) sum += neigh[j];
        if (sum > ans) ans = sum;
    }
    return ans;
};

File: .tmp_rs4013/rust_check_batch08/2497_maximum_star_sum_of_a_graph.rs
struct Solution;
// LeetCode 2497 - Maximum Star Sum of a Graph
// https://leetcode.com/problems/maximum-star-sum-of-a-graph/

impl Solution {
    pub fn max_star_sum(vals: Vec<i32>, edges: Vec<Vec<i32>>, k: i32) -> i32 {
        let n = vals.len();
        let mut g = vec![Vec::new(); n];
        for e in &edges {
            let (a, b) = (e[0] as usize, e[1] as usize);
            g[a].push(b);
            g[b].push(a);
        }
        let mut ans = vals[0];
        for i in 0..n {
            let mut neigh = Vec::new();
            for &v in &g[i] {
                if vals[v] > 0 {
                    neigh.push(vals[v]);
                }
            }
            neigh.sort_unstable_by(|a, b| b.cmp(a));
            let mut sum = vals[i];
            for j in 0..neigh.len().min(k as usize) {
                sum += neigh[j];
            }
            if sum > ans {
                ans = sum;
            }
        }
        ans
    }
}

fn main() {}
